### ISSUE TITLE
Syncing PHPCR permissions and editing permissions fails for some pages after upgrading to 2.3.0

### ISSUE THREAD

**Issue by mareksuscak** [sulu/sulu#6095]
| Q | A
| --- | ---
| Sulu Version | 2.3.0
| PHP Version | 7.4.16
| DB Version | MySQL 8.0.18
| Browser Version | All browsers / Server error

# Actual Behavior

```
# Running the following command fails:
bin/adminconsole sulu:security:sync-phpcr-permissions
```

![image](https://user-images.githubusercontent.com/613647/120463349-1a63e180-c39c-11eb-9ac4-3ed43298353e.png)

Changing permissions for the given page fails:

![image](https://user-images.githubusercontent.com/613647/120463388-28196700-c39c-11eb-8514-e1da8316e560.png)

Error log from Sentry:

![image](https://user-images.githubusercontent.com/613647/120463560-539c5180-c39c-11eb-9ac2-34a3bb533161.png)

HTTP request payload:

```json
{
  "permissions": {
    "1": {
      "view": true,
      "add": true,
      "edit": true,
      "delete": true,
      "archive": true,
      "live": true,
      "security": true
    },
    "2": {
      "view": true,
      "add": true,
      "edit": true,
      "delete": true,
      "archive": false,
      "live": true,
      "security": true
    },
    "3": {
      "view": false,
      "add": false,
      "edit": false,
      "delete": false,
      "archive": false,
      "live": false,
      "security": false
    },
    "4": {
      "view": false,
      "add": false,
      "edit": false,
      "delete": false,
      "live": false,
      "security": false
    }
  }
}
```

# Expected Behavior

The CLI command as well as the permission management screen should work.

# Steps to Reproduce

- Run the CLI command
- Modify permissions for the homepage

# Possible Solutions

n/a

**Comment by mareksuscak** [comment]
I'm opening this issue per request in the community Slack: https://sulu-io.slack.com/archives/C0430GGCV/p1622456802238900

**Comment by alexander-schranz** [comment]
I think there are maybe two cases we did not keep in mind when we created the command:

**Case 1: Role without Permission (phpcr example):**

```bash
| sec:role-2                  | STRING (0)   |                                                                             |
| sec:role-4                  | STRING (4)   | [0] view
```

**Case 2: Removed Role**

We need to check if the roleId really exist in the database bofre try to sync the permissions to the new table.

/cc @luca-rath can you  have a look at this and test this cases?

**Comment by luca-rath** [comment]
@mareksuscak Could you please try, if sulu/sulu#6098 fixes your problem?

**Comment by mareksuscak** [comment]
@luca-rath: That worked! Thanks.

![CleanShot 2021-06-02 at 18 40 55](https://user-images.githubusercontent.com/613647/120518649-1488f300-c3d2-11eb-9d22-080776b4504b.png)

**Comment by alexander-schranz** [comment]
@mareksuscak Thank you for testing this! Its merged and will be part of the 2.3.1 release.

**Comment by luca-rath** [comment]
Closing this, because it has been fixed by #6098
